CMP+CRMF: Re-formulate and (re-)add some comments removed by PR #15539

DDvO opened the pull request:
Those comments had been removed by #15539 because they had the word 'TODO' in them.
They still do have their merits, so I re-formulated them accordingly.

t-j-h commented:
If you have a pile of things which you think should be improved in the code then raise github issues for tracking this.
Comments in the code tend to get completely ignored - and we work off github issues - that is how work gets completed.
Adding these things back into the code will not improve the change that they will be fixed - raising github issues will.

mattcaswell commented:
OTC: Moving to this to Post 3.0 milestone

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 244 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 275 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 306 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 337 days ago

DDvO commented:
Rebased to fix trivial merge conflict.
Ping for reviews. 
This just (re-)adds some comments and moves an aux function to `apps/lib/apps.c`.

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 30 days ago

DDvO commented:
Carved out the code cleanup to #18900.
So this PR now just adds a bunch of comments.

t8m commented:
IMO only the comments which can hint the implementer of a future feature should be added. I.E., do not add comments that are really RFEs by themselves.

DDvO commented:
> IMO only the comments which can hint the implementer of a future feature should be added. I.E., do not add comments that are really RFEs by themselves.

Thanks @t8m for having a look again and form your comments - 
when I have some spare time, I will convert the code comments that hint on feature extensions elsewhere, such as proper FRs.

DDvO commented:
Rebased to fix merge conflict.
Converted one comment to a question: #19805 and another one to a FR: #19087.
Removed, re-formulated, or solved the rest.

DDvO commented:
Please review - it's a pretty low-hanging fruit.

beldmit commented on crypto/crmf/crmf_local.h:
```diff
@@ -329,7 +330,7 @@ struct ossl_crmf_certtemplate_st {
     ASN1_INTEGER *certReqId;
     OSSL_CRMF_CERTTEMPLATE *certTemplate;
-    STACK_OF(OSSL_CRMF_ATTRIBUTETYPEANDVALUE) *controls;
+    STACK_OF(OSSL_CRMF_ATTRIBUTETYPEANDVALUE /* Controls expanded */) *controls;
```
Not sure about style here.

DDvO replied:
At least `check-format.pl crypto/crmf/crmf_local.h` is happy with it.

beldmit reviewed (approved):
LGTM, have one minor style question

DDvO commented:
Merged -  thanks @beldmit and @hlandau.
